### PR TITLE
Flag incompatible add-ons more clearly in restyle list (fix #2024)

### LINKED ISSUE #2024: Add-on not available for your platform UX is less obvious in the refresh.
Old:

featured_extensions____add-ons_for_firefox_for_android

New (with hover applied on the link):

featured_extensions____add-ons_for_firefox_for_android

The refresh version has a lot of whitespace and requires a hover to show the add-on is not relevant. This is all a bit mystery meat. Also if you're only using the keyboard there's no way to infer any information without clicking through so it's a bit of a a11y issue too.

### PR DESCRIPTION
**Pull request by tofumatt**
Fixes #2024; add-ons are now clearly marked as incompatible.

apr 04 2016 10 55

### CODE REVIEW

File: static/css/restyle/restyle.less [hunk 1]
```diff
@@ -1545,11 +1545,26 @@ h3.author .transfer-ownership {
...
   font-size: 14px;
   padding-left: 20px;
+  padding-right: 5px;
   margin-bottom: 5px;
   white-space: normal;
   text-shadow: none;
 }
 
+.items .item.incompatible {
+
```
Comment (EnTeQuAk) [inline static/css/restyle/restyle.less]: We don't seem to do empty lines usually?

File: static/css/restyle/restyle.less [hunk 2]
```diff
@@ -1545,11 +1545,26 @@ h3.author .transfer-ownership {
...
+  .extra .button.not-available {
+    display: none;
+    float: right;
+  }
+
+  &:focus .extra .button.not-available,
+  &:hover .extra .button.not-available {
+    display: block;
+  }
+
```
Comment (EnTeQuAk) [inline static/css/restyle/restyle.less]: Same here

File: static/js/impala/listing.js [hunk 3]
```diff
@@ -32,9 +32,11 @@ function initListingCompat(domContext) {
         var $this = $(this);
         if ($this.find('.acr-override').length) {
             $this.addClass('acr');
-        } else if (!$this.hasClass('ignore-compatibility') &&
-                   $this.find('.concealed').length == $this.find('.button').length) {
-            $this.addClass('incompatible');
+        } else if (!$this.hasClass('ignore-compatibility') && (
+          $this.find('.concealed').length == $this.find('.button').length ||
+          $this.find('button.not-available').length
```
Comment (EnTeQuAk) [inline static/js/impala/listing.js]: I hate if-statements like that but don't mind me not knowing style-guides (if AMO has any) here but I'd usually do something like

```javascript
var isIncompatible = (
    !$this.hasClass('ignore-compatibility') &&
    $this.find('.concealed').length == $this.find('.button').length ||
    $this.find('button.not-available').length);

# ....

} else if isIncompatible
```

but that's just a nitpick.

**Comment by EnTeQuAk** [comment]
Looks great, r+wc

File: static/js/impala/listing.js (hunk 3 above)
Comment (tofumatt) [inline static/js/impala/listing.js]: There's a lot of messy code in here, but I agree. I'll change it up, why not.

**Pull request merged**